src/proc.py

```diff
@@ -29,16 +29,31 @@
 from subprocess import * # flake8: noqa
 
 
-def FixPython(cmd):
+def CanExecute(path):
+  return os.path.isfile(path) and os.access(path, os.X_OK)
+
+
+def Which(filename, cwd):
+  to_search = [cwd] + os.environ['PATH'].split(os.pathsep)
+  for path in to_search:
+    abs_path = os.path.join(path, filename)
+    if CanExecute(abs_path):
+      return abs_path
+  raise Exception('File "%s" not found. (cwd=`%s`, PATH=`%s`' %
+                  (filename, cwd, os.environ['PATH']))
+
+
+def FixPython(cmd, cwd):
   script = cmd[0]
-  if script.endswith('.py') and os.path.exists(script):
-    return [sys.executable] + cmd
+  if script.endswith('.py') and not CanExecute(script):
```

Comment:
I don't think we need to require that python scripts be executable (if you run with `python /path/to/foo.py` it doesn't need to be for it to work). I guess this is sort of the hazard of mixing python handling with executable handling. For executables it's pretty clear what we should do. 
For python, we should at least allow a non-executable script to run if the path given is an absolute path or is a file that exists relative to the cwd. Otherwise I guess we could search the path via Which for the script (including the requirement that it's executable).